utils/economic-checks/src/util.rs

```diff
@@ -0,0 +1,342 @@
+};
+
+use sp_std::collections::btree_map::BTreeMap;
+use std::sync::Arc;
+
+type GasNodeKeyOf<T> = <<T as pallet_gear::Config>::GasHandler as ValueTree>::Key;
+type GasBalanceOf<T> = <<T as pallet_gear::Config>::GasHandler as ValueTree>::Balance;
+
+pub(crate) type WaitlistOf<T> = <<T as pallet_gear::Config>::Messenger as Messenger>::Waitlist;
+
+// Generate a crypto pair from seed.
+pub(crate) fn get_from_seed<TPublic: Public>(seed: &str) -> <TPublic::Pair as Pair>::Public {
+    TPublic::Pair::from_string(&format!("//{}", seed), None)
+        .expect("static values are valid; qed")
+        .public()
+}
+
+type AccountPublic = <Signature as Verify>::Signer;
+
+// Generate an account ID from seed.
+pub(crate) fn get_account_id_from_seed<TPublic: Public>(seed: &str) -> AccountId32
+where
+    AccountPublic: From<<TPublic::Pair as Pair>::Public>,
+{
+    AccountPublic::from(get_from_seed::<TPublic>(seed)).into_account()
+}
+
+// Generate an Aura authority key.
+pub(crate) fn authority_keys_from_seed(s: &str) -> (AccountId32, AuraId, GrandpaId) {
+    (
+        get_account_id_from_seed::<sr25519::Public>(s),
+        get_from_seed::<AuraId>(s),
+        get_from_seed::<GrandpaId>(s),
+    )
+}
+
+pub(crate) fn create_random_accounts(
+    rng: &mut StdRng,
+    root_acc: &AccountId32,
+) -> Vec<(AccountId32, u128)> {
+    let initial_accounts_num = 1 + (rng.next_u32() % 1000); // [1..1000]
+    let mut accounts = vec![(root_acc.clone(), 1_000_000_000_000_000_u128)];
+    for _ in 1..initial_accounts_num {
+        let mut acc_id = [0_u8; 32];
+        rng.fill_bytes(&mut acc_id);
+        let balance = (rng.next_u64() >> 14) as u128; // approx. up to 10^15
+        accounts.push((acc_id.into(), balance));
+    }
+    accounts
+}
+
+pub(crate) fn new_test_ext(
+    balances: Vec<(impl Into<AccountId32>, u128)>,
+    initial_authorities: Vec<(AccountId32, AuraId, GrandpaId)>,
+    root_key: AccountId32,
+) -> sp_io::TestExternalities {
+    let mut t = system::GenesisConfig::default()
+        .build_storage::<Runtime>()
+        .unwrap();
+
+    pallet_balances::GenesisConfig::<Runtime> {
+        balances: balances
+            .into_iter()
+            .map(|(acc, balance)| (acc.into(), balance))
+            .chain(
+                initial_authorities
+                    .iter()
+                    .cloned()
+                    .map(|(acc, _, _)| (acc, 1000)),
+            )
+            .collect(),
+    }
+    .assimilate_storage(&mut t)
+    .unwrap();
+
+    AuraConfig {
+        authorities: initial_authorities
+            .iter()
+            .cloned()
+            .map(|(_, x, _)| x)
+            .collect(),
+    }
+    .assimilate_storage(&mut t)
+    .unwrap();
+
+    BasicExternalities::execute_with_storage(&mut t, || {
+        <GrandpaConfig as GenesisBuild<Runtime>>::build(&GrandpaConfig {
+            authorities: initial_authorities
+                .iter()
+                .map(|x| (x.2.clone(), 1))
```

Comment:
here we already able to call `.into_iter()` to avoid cloning
```suggestion
                .into_iter()
                .map(|x| (x.2, 1))
```

and what this "1" mean?

Reply:
"1" is the `AuthorityWeight` associated with a grandpa authority: https://github.com/paritytech/substrate/blob/master/primitives/finality-grandpa/src/lib.rs#L77